src/tests/Directory.Build.targets

```diff
@@ -153,6 +153,7 @@
   <Target Name="CheckForTestOutsideOfGroup" BeforeTargets="Build">
     <Error Condition="'$(InMergedTestDirectory)' == 'true'
       and '$(OutputType)' == 'Exe'
+      and '$(CLRTestKind)' == 'BuildAndRun'
```

Comment:
It looks like I never had to deal with the kind of case that led to this (in the JIT tests).  I'm a bit concerned because this is the check that we don't exclude a test (say, it was `BuildOnly` with multiple `RunOnly` projects that specify different parameters - which likely should just become `[Theory]s`).  This covers both conversion of tests and authoring new ones (which happened, though this is less likely since test authoring is generally copy-and-paste and we've gotten past conversion of most tests).

On the other hand, `BuildOnly` is exactly how one should specify "this is a weird test - I will take care of running it".

Perhaps auditing is sufficient for these (easy here - I only see ParallelCrash* and stackoverflow* - it will be something to keep in mind, especially in GC)

Reply:
I just retried and the check seems to be no longer needed so I removed it. I guess I was hitting the problem in some of the internal test dependencies that should specify ReferenceXUnitWrapperGenerator=false instead. Thanks for pointing that out!